### DISCUSSION TITLE
Payment.status should have additional enum for Pending

### DISCUSSION THREAD

**Discussion by BLR-0118** [beckn/protocol-specifications#279] [Core Specification]
Payment.status has 2 enums - PAID, NOT-PAID. Often payment is made but confirmation gets delayed for which an additional status 'PENDING' may be considered

**Comment by Vedsaga** [comment]
Another status should be also there, call `partially` so, when it comes to B2B transaction... there are instances when many times some amounts are pre-paid prior to the order and some payment are paid later... similarly in B2C transaction it could happen that consumer paid 50% prior to the order or reaming later...

**Comment by ravi-prakash-v** [comment]
@BLR-0118 from the POV of BPP, it is the BAP that needs to pay it. So even if the BAP has collected the money, from the pov of the BPP, it is still NOT-PAID. BPP doesn't care whether the BAP has collected the money or not. BPP only cares if BAP has settled the money or not.  

The `Payment.status` field only captures the settlement status between the BAP and the BPP. So IMHO, I don't think a PENDING status is required. I think the `collected_by` field and the `status` field is more than enough to capture all settlement scenarios between the BAP and the BPP. Hope that clears things up. 

@pramodkvarma  @venkatramanm any thoughts on this?

**Comment by BLR-0118** [comment]
There could be a situation where the buyer or seller requires prepaid payment for order to go through. Hence, if payment has been made but confirmation not received, this additional status will be useful in communicating to the counterparty the current status.

**Comment by ravi-prakash-v** [comment]
@BLR-0118 Not sure what "order to go through" means. Does it mean,

a) Order has been placed by the buyer on the BAP app? OR,
b) Order has been received by the BPP but has not been confirmed yet? OR,
b) Order has been confirmed at the BPP? OR, 
c) Confirmed order has been received at the BAP from the BPP? OR,
e) Something else?

Second ambiguity in the above statement is around the phrase "payment has to be _made_". What does this mean? Is it,

a) Payment has to be collected from the user? OR,
b) Payment has to be settled to the BPP? OR, 
c) Payment has to be credited to the BPP's seller account? OR,
d) Something else?

Third ambiguity is in the definition of "counterparty". Who is the counterparty here? And from who's point of view are we talking here?

a) BPP from the pov of the seller? OR,
b) BAP from the pov of the BPP? OR,
c) Buyer from the pov of the BAP? OR,
d) BAP from the pov of the seller? OR,
e) Buyer from the pov of the seller?

In any case, the Payment.status only contains the settlement status between the BAP and the BPP. This value has "NO" influence on what the end-customer sees on their app. If the BAP collects money from the user first and _then_ settles it with the BPP on a T+1 basis, the user will see payment status as "PAID" but the order object will will show `Payment.status` as `NOT-PAID`. 

Once the BAP settles the money to the BPP after 1 day, the `Payment.status` value will be set to `PAID` by the BPP. Until it receives an order object with payment.status equal to `"PAID"`, it should periodically fire a `status` call to the BPP to fetch the latest status of the order. 

But again, the above interaction doesn't influence the user experience on the app. So, either the money is settled with the BPP, or is not settled. Hence, not sure why a `PENDING` state is needed here.

**Comment by pramodkvarma** [comment]
Since payment is out of band (not a core beckn capability, separately processed), various payment statuses may be necessary in reality (partial payments, payment done-but-not hit the provider, etc). 

Before we incorporate, I am also curious on the immediate need within current pilot networks. I am also curious who is updating the status to `PENDING` etc within the network.
